### PR TITLE
feat(spanner): add samples for database roles

### PR DESCRIPTION
**Pull request by varuncnaik**
Adds samples with database roles on requests.

### CODE REVIEW

**Comment by varuncnaik** [comment]
I'll work on a sample for ListDatabaseRoles soon.

**Review by enocom — APPROVED** [review]

**Comment by enocom** [comment]
@varuncnaik Looks like there are some compilation errors caused by missing types. You'll need to upgrade the underlying client if these are new methods and types you're using.

**Comment by enocom** [comment]
@varuncnaik Still seeing some compilation errors. Let me know if you'd like some help with this. Otherwise, I'll assume you ping me when you're ready for another look.

**Comment by enocom** [comment]
Looks like we're still missing some types:

```
# github.com/GoogleCloudPlatform/golang-samples/spanner/spanner_snippets
spanner_snippets/snippet.go:825:3: unknown field 'DatabaseRole' in struct literal of type "cloud.google.com/go/spanner".ClientConfig
# github.com/GoogleCloudPlatform/golang-samples/spanner/spanner_snippets/spanner
spanner_snippets/spanner/spanner_read_data_with_database_role.go:32:3: unknown field 'DatabaseRole' in struct literal of type "cloud.google.com/go/spanner".ClientConfig
```

@varuncnaik Want some help finishing this? We've had this open for quite awhile now.

**Comment by varuncnaik** [comment]
Hi @enocom, we're aware of the missing types. In my local setup for this PR, I'm editing my go.mod file to replace cloud.google.com/go/spanner with my local copy of https://github.com/googleapis/google-cloud-go/pull/5701.

We plan to merge both of these PRs around the time that Cloud Spanner fine-grained access control becomes generally available. Until then, I am periodically merging the latest `main` into both PRs and making sure that tests pass locally. CC @rahul2393, in case the two of you want to exchange information over DM.

**Comment by enocom** [comment]
Sounds good. Just ping me when you're ready and we can merge this.

File: spanner/spanner_snippets/spanner/spanner_add_and_drop_database_roles.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+// [START spanner_add_and_drop_database_roles]
```
Comment (rahul2393) [inline spanner/spanner_snippets/spanner/spanner_add_and_drop_database_roles.go:17]:
```suggestion
// [START spanner_add_and_drop_database_roles]
```
// [START spanner_add_and_drop_database_role]

File: spanner/spanner_snippets/spanner/spanner_add_and_drop_database_roles.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	if err != nil {
+		return err
+	}
+	if err := op.Wait(ctx); err != nil {
+		return err
+	}
+	return nil
+}
+
+// [END spanner_add_and_drop_database_roles]
```
Comment (rahul2393) [inline spanner/spanner_snippets/spanner/spanner_add_and_drop_database_roles.go:70]:
```suggestion
// [END spanner_add_and_drop_database_role]
```

**Review by rahul2393 — CHANGES_REQUESTED** [review]
LGTM with minor change.

**Review by rahul2393 — APPROVED** [review]

**Pull request merged**